python-frontend/src/main/java/org/sonar/python/metrics/FileLinesVisitor.java

```diff
@@ -138,14 +138,19 @@ private void visitToken(Token token) {
   private void visitComment(Trivia trivia, Token parentToken) {
     String commentLine = getContents(trivia.token().value());
     int line = trivia.token().line();
-    if (commentLine.contains("NOSONAR")) {
+    if (containsNoSonarComment(trivia)) {
       linesOfComments.remove(line);
       addNoSonarLines(trivia, parentToken);
     } else if (!isBlank(commentLine)) {
       linesOfComments.add(line);
     }
   }
 
+  public static boolean containsNoSonarComment(Trivia trivia){
+    String commentLine = getContents(trivia.token().value());
+    return commentLine.contains("NOSONAR");
+  }
```

Comment:
I'm not sure if it's really necessary to create a public method here. Especially when your check accesses the helper method from a non-check-related class like this visitor. I would leave the FileLineVisitor as it is and just duplicate the one-line logic in the check.

Reply:
It has been discussed and, while the location of this method is indeed not the best, we've chosen to keep it there for the moment to have a consistency when detecting the NOSONAR marker as it is also used in visitComment() right above.